plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -118,8 +143,64 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle)
             if (min != null && max != null) {
                 columnBuilder.setRange(DoubleRange.from(columnHandle.getType(), min, max));
             }
+            columnBuilder.setDistinctValuesCount(
+                    Optional.ofNullable(ndvs.get(fieldId))
+                            .map(Estimate::of)
+                            .orElseGet(Estimate::unknown));
             columnHandleBuilder.put(columnHandle, columnBuilder.build());
         }
         return new TableStatistics(Estimate.of(recordCount), columnHandleBuilder.buildOrThrow());
     }
+
+    private Map<Integer, Long> readNdvs(ConnectorSession session, Table icebergTable)
+    {
+        Map<Integer, Long> ndvByColumnId = new HashMap<>();
+        try (FileIO fileIo = fileIoProvider.createFileIo(new HdfsContext(session), session.getQueryId())) {
+            getStatisticsFiles(icebergTable)
+                    .filter(statisticsFile -> statisticsFile.statistics().containsKey(NDV_STATS))
+                    // newest first
+                    .sorted(comparing(StatisticsFile::sequenceNumber).reversed())
+                    .forEach(statisticsFile -> {
+                        Set<List<Integer>> columnSets = statisticsFile.statistics().get(NDV_STATS);
+                        Set<Integer> columnIds = columnSets.stream()
+                                .filter(columnSet -> columnSet.size() == 1)
```

Comment:
When would they have more than 1 element?

Reply:
You could calculate ndv of tuples (eg to cater for correlations).
we don't support this, so we filter